internal/services/loganalytics/log_analytics_workspace_resource.go

```diff
@@ -379,11 +350,151 @@ func resourceLogAnalyticsWorkspaceCreateUpdate(d *pluginsdk.ResourceData, meta i
 	return resourceLogAnalyticsWorkspaceRead(d, meta)
 }
 
+func resourceLogAnalyticsWorkspaceUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).LogAnalytics.WorkspaceClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+	log.Printf("[INFO] preparing arguments for AzureRM Log Analytics Workspace creation.")
+
+	var isLACluster bool
+
+	id, err := workspaces.ParseWorkspaceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	existing, err := client.Get(ctx, *id)
+	if err != nil {
+		return fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	if existing.Model == nil {
+		return fmt.Errorf("retrieving %s: `model` was nil", *id)
+	}
+
+	if existing.Model.Properties == nil {
+		return fmt.Errorf("retrieving %s: `model.Properties` was nil", *id)
+	}
+	payload := existing.Model
+	props := payload.Properties
+
+	if d.HasChange("allow_resource_only_permissions") {
+		if props.Features == nil {
+			props.Features = &workspaces.WorkspaceFeatures{}
+		}
+		props.Features.EnableLogAccessUsingOnlyResourcePermissions = pointer.To(d.Get("allow_resource_only_permissions").(bool))
+	}
+
+	if d.HasChange("local_authentication_disabled") {
+		if props.Features == nil {
+			props.Features = &workspaces.WorkspaceFeatures{}
+		}
+		props.Features.DisableLocalAuth = pointer.To(d.Get("local_authentication_disabled").(bool))
+	}
+
+	if d.HasChange("cmk_for_query_forced") {
+		props.ForceCmkForQuery = pointer.To(d.Get("cmk_for_query_forced").(bool))
+	}
+
+	if d.HasChange("identity") {
+		expandedIdentity, err := identity.ExpandSystemOrUserAssignedMap(d.Get("identity").([]interface{}))
+		if err != nil {
+			return fmt.Errorf("expanding `identity`")
+		}
+		payload.Identity = expandedIdentity
+	}
+
+	if d.HasChange("internet_ingestion_enabled") {
+		props.PublicNetworkAccessForIngestion = pointer.To(workspaces.PublicNetworkAccessTypeDisabled)
+		if d.Get("internet_ingestion_enabled").(bool) {
+			props.PublicNetworkAccessForIngestion = pointer.To(workspaces.PublicNetworkAccessTypeEnabled)
+		}
+	}
+
+	if d.HasChange("internet_query_enabled") {
+		props.PublicNetworkAccessForQuery = pointer.To(workspaces.PublicNetworkAccessTypeDisabled)
+		if d.Get("internet_query_enabled").(bool) {
+			props.PublicNetworkAccessForQuery = pointer.To(workspaces.PublicNetworkAccessTypeEnabled)
+		}
+	}
+
+	skuName := d.Get("sku").(string)
+	if sku := props.Sku; sku != nil {
+		if strings.EqualFold(string(sku.Name), string(workspaces.WorkspaceSkuNameEnumLACluster)) {
+			isLACluster = true
+		}
+	}
```

Comment:
should we move this inside the `if d.HasChange("sku")` check since it doesn't look like this is getting used anywhere else?